administrator/components/com_menus/src/Field/MenuItemByComponentField.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_menus
+ *
+ * @copyright   (C) 2022 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\Component\Menus\Administrator\Field;
+
+use Joomla\CMS\Factory;
+use Joomla\CMS\Form\Field\ListField;
+use Joomla\CMS\Language\Text;
+use Joomla\Database\ParameterType;
+use Joomla\Utilities\ArrayHelper;
+
+/**
+ * MenuItem by Component field.
+ *
+ * @since __DEPLOY_VERSION__
+ */
+class MenuItemByComponentField extends ListField
+{
+    /**
+     * The form field type.
+     *
+     * @var     string
+     * @since   __DEPLOY_VERSION__
+     */
+    protected $type = 'MenuItemByComponent';
+
+    /**
+     * Method to get a list of options for a list input.
+     *
+     * @return    array  An array of JHtml options.
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    protected function getOptions()
+    {
+        // Initialise variable.
+        $db      = Factory::getDbo();
```

Comment:
```suggestion
        $db      = $this->getDatabase();
```
See https://github.com/joomla/joomla-cms/pull/37097